### PR TITLE
Convert play-mvc-2.6 groovy to java

### PR DESCRIPTION
**Pull request by heyams**

### CODE REVIEW

**Comment by heyams** [comment]
i expect this PR to be green after the CI build run :)
please review when you can.

**Review by jaydeluca — APPROVED** [review]

File: instrumentation/play/play-mvc/play-mvc-2.6/javaagent/build.gradle.kts [hunk 1]
```diff
@@ -60,19 +60,21 @@ testing {
...
-      enabled = false
-    }
+  }
 
-    check {
-      dependsOn(testing.suites)
-    }
+  named("latestDepTest") {
+    enabled = testLatestDeps
+  }
```
Comment (laurit) [inline instrumentation/play/play-mvc/play-mvc-2.6/javaagent/build.gradle.kts:74]: why is this needed?

File: instrumentation/play/play-mvc/play-mvc-2.6/javaagent/src/latestDepTest/java/io/opentelemetry/javaagent/instrumentation/play/v2_6/PlayServerTest.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  @Override
+  protected void stopServer(Server server) {
+    server.stop();
+  }
+
+  @Override
+  protected void configure(HttpServerTestOptions options) {
+    super.configure(options);
+    options.setHasHandlerSpan(unused -> true);
+    options.setTestHttpPipelining(false);
```
Comment (laurit) [inline instrumentation/play/play-mvc/play-mvc-2.6/javaagent/src/latestDepTest/java/io/opentelemetry/javaagent/instrumentation/play/v2_6/PlayServerTest.java:106]: in groovy test this was only disabled for `PlayAsyncServerTest`

File: instrumentation/play/play-mvc/play-mvc-2.6/javaagent/build.gradle.kts (hunk 1 above)
Comment (heyams) [reply instrumentation/play/play-mvc/play-mvc-2.6/javaagent/build.gradle.kts:74]: it wasn't triggering latestDepTest module test when i have -PtestLatestDeps=true locally.

Comment (laurit) [reply instrumentation/play/play-mvc/play-mvc-2.6/javaagent/build.gradle.kts:74]: you sure about that? seems to work fine for me, Perhaps nothing was run because you had already run the tests and got the results from build cache?

Comment (heyams) [reply instrumentation/play/play-mvc/play-mvc-2.6/javaagent/build.gradle.kts:74]: let me 'invalidate cache' and try it without this.

**Comment by heyams** [comment]
will close and reopen to trigger CI builds rerun. errors are not related to my change.
